### PR TITLE
Switch to MMCV MODEL_REGISTRY

### PR DESCRIPTION
**Pull request by Yshuo-Li**

### CODE REVIEW

File: mmedit/models/builder.py [hunk 1]
```diff
@@ -1,24 +1,11 @@
-import torch.nn as nn
-from mmcv import build_from_cfg
+from mmcv.cnn import build_model_from_cfg
+from mmcv.utils import Registry
 
-from .registry import BACKBONES, COMPONENTS, LOSSES, MODELS
+MODELS = Registry('models', build_func=build_model_from_cfg)
```
Comment (nbei) [inline mmedit/models/builder.py:4]: You may refer to [MMDet example](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/builder.py#L6) to set `parent` argument with MMCV_MODELS.

**Comment by innerlee** [comment]
@Yshuo-Li ci fails

File: mmedit/datasets/base_matting_dataset.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def __init__(self, ann_file, pipeline, data_prefix=None, test_mode=False):
         self.data_prefix = str(data_prefix)
         self.data_infos = self.load_annotations()
 
-    def evaluate(self, results, logger=None):
+    def evaluate(self, results):
```
Comment (innerlee) [inline mmedit/datasets/base_matting_dataset.py:18]: Reason for this change?

File: mmedit/datasets/registry.py [hunk 3]
```diff
@@ -1,4 +1,12 @@
...
-PIPELINES = Registry('pipeline')
+from .builder import DATASETS, PIPELINES
+
+__all__ = ['DATASETS', 'PIPELINES']
+
+warnings.simplefilter('once', DeprecationWarning)
+warnings.warn(
+    'Registries (DATASETS, PIPELINES) have been moved to '
+    'mmedit.datasets.builder. Importing from '
+    'mmedit.models.registry will be deprecated in the future.',
```
Comment (innerlee) [inline mmedit/datasets/registry.py:11]: add pr link in the warning `details see https://github.com/open-mmlab/mmediting/pull/446`

File: mmedit/models/registry.py [hunk 4]
```diff
@@ -1,6 +1,12 @@
...
+from .builder import BACKBONES, COMPONENTS, LOSSES, MODELS
+
+__all__ = ['BACKBONES', 'COMPONENTS', 'LOSSES', 'MODELS']
+
+warnings.simplefilter('once', DeprecationWarning)
+warnings.warn(
+    'Registries (BACKBONES, COMPONENTS, LOSSES, MODELS) have '
+    'been moved to mmedit.models.builder. Importing from '
+    'mmedit.models.registry will be deprecated in the future.',
+    DeprecationWarning)
```
Comment (innerlee) [inline mmedit/models/registry.py:12]: same as above

File: mmedit/datasets/base_matting_dataset.py (hunk 2 above)
Comment (Yshuo-Li) [reply mmedit/datasets/base_matting_dataset.py:18]: Because I have found that `logger` has not been used in this function.

Comment (innerlee) [reply mmedit/datasets/base_matting_dataset.py:18]: The caller might pass logger into

**Comment by innerlee** [comment]
@zhouzaida would like to take a quick look?

**Comment by wangruohui** [comment]
Any relation to this newer PR? https://github.com/open-mmlab/mmediting/pull/660

**Pull request closed without merging**